frameworks/Java/gemini/gemini-mysql.dockerfile

```diff
@@ -0,0 +1,18 @@
+FROM resin:latest
+
+RUN apt-get install -qqy -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" \
+    ant
+
+ADD Docroot/ /gemini/Docroot
+ADD Source/ /gemini/Source
+ADD build.xml /gemini/
+ADD ivy.xml /gemini/
+ADD ivysettings.xml /gemini/
+
+RUN cd /gemini/Docroot/WEB-INF; mv gemini-mysql.conf GeminiHello.conf;
+
+RUN cd /gemini; mkdir -p Docroot/WEB-INF/classes; mkdir -p Docroot/WEB-INF/lib; ant resolve; ant compile
+
+EXPOSE 8080
+
+CMD ["resinctl", "-conf", "/gemini/Docroot/WEB-INF/resin.xml", "console"]
```

Comment:
What is our general guideline for when to use `RUN` or `CMD`?  Is it:  Use `RUN` for every bash shell command involved in setup except the last one, and use `CMD` for the last bash command that launches the server?

Reply:
`RUN` is only used for setup; it will not actually run and persist a process. See [this comment](https://github.com/TechEmpower/FrameworkBenchmarks/pull/3292/files#diff-f4a364bcc831a5afd517148bd6ce7415R31) for an example.

`CMD` is what is executed when we start the container. This version does not execute a shell, but it could have just as well been `CMD resin -conf /gemini/Docroot/WEB-INF/resin.xml console` and that would have invoked `/bin/sh resin -conf ...` (the reaper will clean this either way. For instances where we need to run multiple processes (like PHP using `php-fpm` and `nginx`) we should be able to get away with `CMD php-fpm ... && nginx`